clients/java/signalr/src/main/java/com/microsoft/signalr/HubConnection.java

```diff
@@ -32,13 +34,13 @@
     private Logger logger;
     private List<Consumer<Exception>> onClosedCallbackList;
     private boolean skipNegotiate;
-    private Supplier<CompletableFuture<String>> accessTokenProvider;
+    private Supplier<Single<String>> accessTokenProvider;
```

Comment:
We can just use `Single<String>` instead of `Supplier<Single<String>>` because each subscription to the Single re-invokes whatever the user provided as their implementation.

For example, I could implement the "renew token when expired" logic using pseudo-code like this:

```java
builder.withAccessTokenProvider(Single.fromCallable(() -> {
    if(accessToken == null || accessToken.isExpired()) {
        accessToken = renewAccessToken();
    }
    return accessToken;
});
```